### ISSUE TITLE
RIDE should add ${EMPTY} to cells before first comment cell

### ISSUE THREAD

**Issue by yanne** [robotframework/RIDE#922]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=886) by alejandr... on 1 Nov 2011*

The placement of a comment should never produce a failure. Consider the case of a keyword with an optional parameter which can't be empty:

|----------------------------------------------------------------|
| Keyword With Optional Parameter | # This PASSes |              |
|---------------------------------|---------------|--------------|
| Keyword With Optional Parameter |               | # This FAILs |
|----------------------------------------------------------------|

Both scenarios should pass, as the user should be able to add comments wherever is more convenient.

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=886#c1) by @mkorpela on 1 Nov 2011*


In the second version you are overwriting the optional argument value with empty "" - thus the lines have different meaning. 
Could this be the reason why the second version fails in your case?

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=886#c2) by alejandr... on 15 Nov 2011*


Hi, sorry it took so long to reply

You are right, the second version fails because the optional parameter is implicitly set to ${EMPTY}, and the keyword doesn't know how to handle it.

The point I was trying to make is that either RIDE or RobotFramework should have the smarts to deal with this scenario by not passing the ${EMPTY} at all, because it results in unexpected failures.

A more viable workaround could be to immediately set the contents of the cell to ${EMPTY}, so the user is aware of what's going on. Currently, RIDE shows that the cell has been set to ${EMPTY} only after saving and reloading the file.

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=886#c4) by @yanne on 2 Jan 2012*


I think that we could show the explicit ${EMPTY}s in RIDE in this case. 

The original problem cannot be fixed in RIDE, since it relates to how Robot Framework parses data. I am not sure whether that can be changed either, but if you feel this is an important issue, you can open in at the RF issue tracker.

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=886#c5) by @pekkaklarck on 5 Jan 2012*


For Robot itself both of the original examples are equivalent. You can test that, for example, by running the following test case. If the empty cell wasn't ignored the test would fail.

*** Test Cases ***
Example
|  | Log | Hello |   | # Empty cell before is ignored |

If RIDE adds ${EMPTY} to the data in this case that's definitely a bug. I tried to reproduce that with the current version from the version control system but couldn't. Could the bug perhaps been fixed? I target this for 0.40 so that we can test this next week.

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=886#c6) by @yanne on 9 Jan 2012*


We'll decided not to investigate this more for 0.40. 

If this bug is still present in 0.40, we can try to fix this.

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=886#c7) by @mkorpela on 15 Feb 2012*


As peke noted there are situations when ${EMPTY} can not be added. (in case we have arguments with default value).

Closing as WontFix.
